src/components/data/listing/UploadDropTarget.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @author johnworth
+ *
+ * Component that allows files to be uploaded when they are dropped on child components.
+ *
+ * @module UploadDropTarget
+ */
+
+import React, { useState } from "react";
+import processDroppedFiles, { startUpload } from "./UploadDrop";
+import { useUploadTrackingDispatch } from "../../../contexts/uploadTracking";
+import PropTypes from "prop-types";
+
+/**
+ * Utility function that prevents an event from propagating or using a
+ * default handler.
+ *
+ * @param {object} event - The event that got fired off.
+ * @returns null
+ */
+const setupEvent = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+};
+
+/**
+ * @typedef UploadDropTargetProps
+ * @property {Object} children - Child render props.
+ * @property {string} path - The upload directory in the data store.
+ * @property {number} uploadsCompleted - The number of uploads that have completed.
+ * @property {func} setUploadsCompleted - Function that updates uploadsCompleted.
+ */
+
+/**
+ * Performs file uploads to the `path` directory in the data store when
+ * files are dropped onto the child components.
+ *
+ * @param {UploadDropTargetProps} props - The props for the component.
+ * @returns {Object}
+ */
+const UploadDropTarget = (props) => {
+    const uploadDispatch = useUploadTrackingDispatch();
+    const [dragCounter, setDragCounter] = useState(0);
+    const { children, path, uploadCompletedCB } = props;
```

Comment:
It took me a second to parse `CB`.
I think I'm used to seeing callbacks named something like `onUploadCompleted`, but it's not a problem.